code/modules/projectiles/ammunition.dm

```diff
@@ -45,6 +45,39 @@
 		else
 			to_chat(user, "<font color='blue'>You inscribe \"[label_text]\" into \the [initial(BB.name)].</font>")
 			BB.name = "[initial(BB.name)] (\"[label_text]\")"
+	else if(istype(I, /obj/item/ammo_magazine) && isturf(loc)) // Mass magazine reloading.
+		var/obj/item/ammo_magazine/box = I
+		if (!box.can_remove_ammo || box.reloading)
+			return ..()
+
+		box.reloading = TRUE
+		var/boolets = 0
+		var/turf/floor = loc
+		for(var/obj/item/ammo_casing/bullet in floor)
+			if(box.stored_ammo.len >= box.max_ammo)
+				break
+			if(box.caliber == bullet.caliber && bullet.BB)
+				if (boolets < 1)
+					to_chat(user, "<span class='notice'>You start collecting shells.</span>") // Say it here so it doesn't get said if we don't find anything useful.
+				if(do_after(usr,5,box))

```

Comment:
`user` should be valid here, it's best practice to avoid using usr where possible

Reply:
Whoops, that one slipped past my attention.